### PR TITLE
MCP-2730 Claim-Metrics getHealthEvidence

### PR DESCRIPTION
**Pull request by ianwolson**
## What was the problem?

Associated tickets or Slack threads:

- [MCP-2730](https://amida.atlassian.net/browse/MCP-2730)
- [MCP-2371](https://amida.atlassian.net/browse/MCP-2731)

## How does this fix it?[^1]
Removed the old v2/health-data-assessment and added a v2/health-evidence endpoint to Claim Metrics.  This endpoint uses the v2 health evidence collection process from MasIntegrationRoutes to provide a health assessment for a claim.  This endpoint does not write anything to the database.

## How to test this PR
- Step 1
- Step 2[^secrel]

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

**Review by au2 — CHANGES_REQUESTED** [review]
v2/health-data-assessment was not as useful because it uses the v1 logic and do not include conditions. if possible switch to what v2 assessment uses.

File: api/src/main/java/gov/va/vro/api/resources/ClaimMetricsResource.java [hunk 1]
```diff
@@ -149,4 +151,34 @@ ResponseEntity<List<ExamOrderInfoResponse>> allExamOrderInfo(
       @RequestParam(name = "size", required = false, defaultValue = "10")
           @Min(value = 1, message = "invalid size")
           Integer size);
+
+  @Operation(
+      summary =
+          "Retrieves health evidence for a specific claimSubmissionId, veteran ICN, and daignostic code.",
+      description =
+          "This endpoint does the same thing v2/health-data-assessment used to do. "
```
Comment (hayleymdawson) [inline api/src/main/java/gov/va/vro/api/resources/ClaimMetricsResource.java:159]: we probably need to support v2 and v1 through the same endpoint.

File: controller/src/main/java/gov/va/vro/controller/ClaimMetricsController.java [hunk 2]
```diff
@@ -74,4 +84,47 @@ public ResponseEntity<List<ExamOrderInfoResponse>> allExamOrderInfo(Integer page
...
+    if (diagnosis == null) {
+      throw new ClaimProcessingException(
+          claim.getDiagnosticCode(),
+          HttpStatus.BAD_REQUEST,
+          String.format(
+              "Claim with [diagnosticCode = %s] is not in scope.", claim.getDiagnosticCode()));
+    }
+    try {
+      Claim model = postClaimRequestMapper.toModel(claim);
+      model.setIdType(MasAutomatedClaimPayload.CLAIM_V2_ID_TYPE);
```
Comment (hayleymdawson) [inline controller/src/main/java/gov/va/vro/controller/ClaimMetricsController.java:105]: can we repeat the pattern of default to v2, ask for v1? I will confirm with Yoom today about removing the other two endpoints.

File: app/src/test/java/gov/va/vro/controller/VroControllerTest.java [hunk 3]
```diff
@@ -71,141 +64,147 @@ class VroControllerTest extends BaseControllerTest {
...
-    headers.put("content-type", "application/json");
-    var responseEntity =
-        post(
-            "/v2/health-data-assessment",
-            "{ \"one\":\"one\", \"two\":\"two\",}",
-            headers,
-            ClaimProcessingError.class);
-    assertEquals(HttpStatus.BAD_REQUEST, responseEntity.getStatusCode());
-  }
+  //  @Test
```
Comment (hayleymdawson) [inline app/src/test/java/gov/va/vro/controller/VroControllerTest.java:67]: v1 endpoint has to be untouched so this needs to come back. Which may impact needing a default in the HealthAssessment Request.

File: controller/src/main/java/gov/va/vro/controller/ClaimMetricsController.java [hunk 4]
```diff
@@ -74,4 +85,24 @@ public ResponseEntity<List<ExamOrderInfoResponse>> allExamOrderInfo(Integer page
     ExamOrdersInfo examOrdersInfo = claimMetricsService.findAllExamOrderInfo(params);
     return ResponseEntity.ok(examOrdersInfo.getExamOrderInfoList());
   }
+
+  @Override
+  public ResponseEntity<HealthDataAssessment> healthEvidence(MasAutomatedClaimRequest request) {
+    log.info("Received health evidence request with collection ID {}", request.getCollectionId());
+    String correlationId = UUID.randomUUID().toString();
+    var payload =
+        MasAutomatedClaimPayload.builder()
```
Comment (hayleymdawson) [inline controller/src/main/java/gov/va/vro/controller/ClaimMetricsController.java:94]: There should be a mapper, but maybe it isn't accessible from here? if not, i'd put in a tech debt ticket about it and i can take a look next week.

File: service/provider/src/main/java/gov/va/vro/service/provider/camel/PrimaryRoutes.java [hunk 5]
```diff
@@ -17,6 +17,7 @@
 @RequiredArgsConstructor
 public class PrimaryRoutes extends RouteBuilder {
   public static final String ENDPOINT_SUBMIT_CLAIM_FULL = "direct:claim-submit-full";
+  public static final String ENDPOINT_GET_HEALTH_EVIDENCE = "direct:get-health-evidence";
```
Comment (hayleymdawson) [inline service/provider/src/main/java/gov/va/vro/service/provider/camel/PrimaryRoutes.java:20]: This looks to be unused (and should be unused?)

**Review by hayleymdawson — CHANGES_REQUESTED** [review]
Remove Primary Route changes if they aren't used (if they are, let me know why as I'm not seeing it).
Bring back v1 tests and the v1 endpoint has to merge.

Logic for the new mas integration route looks good to me, but obviously afsin will need to chime in.

File: api/src/main/java/gov/va/vro/api/resources/ClaimMetricsResource.java [hunk 6]
```diff
@@ -149,4 +155,41 @@ ResponseEntity<List<ExamOrderInfoResponse>> allExamOrderInfo(
...
+        @ApiResponse(
+            responseCode = "500",
+            description = "Internal service error",
+            content = @Content(schema = @Schema(hidden = true)))
+      })
+  @ResponseStatus(HttpStatus.OK)
+  @Timed(value = "health-evidence")
+  @Tag(name = "Claim Metrics")
+  @ResponseBody
+  ResponseEntity<HealthDataAssessment> healthEvidence(
```
Comment (hayleymdawson) [inline api/src/main/java/gov/va/vro/api/resources/ClaimMetricsResource.java:187]: Also i'm not sure if the HDA object here has everything we want. you may need FullHealthDataAssessmentResponse instead to expose the sufficientEvidence flag.

**Comment by ianwolson** [comment]
Closing and making fresh PR here https://github.com/department-of-veterans-affairs/abd-vro/pull/1487

**Pull request closed without merging**